File: Baekjun/8.3/18185. 라면 사기 (Small)/solution.py
N = int(input())
factory = list(map(int, input().split()))
factory.append(0)
factory.append(0)

ans = 0

for i in range(N):
    if factory[i + 1] > factory[i + 2]:
        c5 = min(factory[i], factory[i + 1] - factory[i + 2])
        factory[i] -= c5
        factory[i + 1] -= c5
        ans += c5 * 5

        c7 = min(factory[i], factory[i + 1], factory[i + 2])
        factory[i] -= c7
        factory[i + 1] -= c7
        factory[i + 2] -= c7
        ans += c7 * 7
    else:
        c7 = min(factory[i], factory[i + 1])
        factory[i] -= c7
        factory[i + 1] -= c7
        factory[i + 2] -= c7
        ans += c7 * 7

        c5 = min(factory[i], factory[i + 1])
        factory[i] -= c5
        factory[i + 1] -= c5
        ans += c5 * 5

    ans += factory[i] * 3

print(ans)

File: Baekjun/8.3/18185. 라면 사기 (Small)/solution.swift
if let N = Int(readLine()!) {
    let buf: [Int] = readLine()!.split(separator:" ").compactMap {Int($0)}
    var arr: [Int] = Array(repeating: 0, count: N+2)
    var ans = 0
  
    for i in 0..<N {
      arr[i] = buf[i]
    }
  
    for i in 0..<N {
      if arr[i+1] > arr[i+2] {
        var c5 = min(arr[i], arr[i+1] - arr[i+2])
        ans += c5 * 5
        arr[i] -= c5
        arr[i + 1] -= c5

        var c7 = min(arr[i], arr[i+1], arr[i+2])
        ans += c7 * 7
        arr[i] -= c7
        arr[i+1] -= c7
        arr[i+2] -= c7
      } else {
        var c7 = min(arr[i], arr[i+1])
        ans += c7 * 7
        arr[i] -= c7
        arr[i+1] -= c7
        arr[i+2] -= c7

        var c5 = min(arr[i], arr[i+1])
        ans += c5 * 5
        arr[i] -= c5
        arr[i+1] -= c5
        
      }
      ans += 3*arr[i]
    }

  print(ans)
}